--- pspell.c	2012-08-15 12:26:05.000000000 +0800
+++ pspell.patched.c	2012-09-11 23:15:12.051458150 +0800
@@ -65,6 +65,7 @@
 static PHP_FUNCTION(pspell_config_create);
 static PHP_FUNCTION(pspell_config_runtogether);
 static PHP_FUNCTION(pspell_config_mode);
+static PHP_FUNCTION(pspell_config_set);
 static PHP_FUNCTION(pspell_config_ignore);
 static PHP_FUNCTION(pspell_config_personal);
 static PHP_FUNCTION(pspell_config_dict_dir);
@@ -145,6 +146,12 @@
 	ZEND_ARG_INFO(0, mode)
 ZEND_END_ARG_INFO()
 
+ZEND_BEGIN_ARG_INFO_EX(arginfo_pspell_config_set, 0, 0, 3)
+        ZEND_ARG_INFO(0, conf)
+        ZEND_ARG_INFO(0, option)
+        ZEND_ARG_INFO(0, value)
+ZEND_END_ARG_INFO()
+
 ZEND_BEGIN_ARG_INFO_EX(arginfo_pspell_config_ignore, 0, 0, 2)
 	ZEND_ARG_INFO(0, conf)
 	ZEND_ARG_INFO(0, ignore)
@@ -192,6 +199,7 @@
 	PHP_FE(pspell_config_create,		arginfo_pspell_config_create)
 	PHP_FE(pspell_config_runtogether,	arginfo_pspell_config_runtogether)
 	PHP_FE(pspell_config_mode,			arginfo_pspell_config_mode)
+        PHP_FE(pspell_config_set,                      arginfo_pspell_config_set)
 	PHP_FE(pspell_config_ignore,		arginfo_pspell_config_ignore)
 	PHP_FE(pspell_config_personal,		arginfo_pspell_config_personal)
 	PHP_FE(pspell_config_dict_dir,		arginfo_pspell_config_dict_dir)
@@ -795,6 +803,35 @@
 }
 /* }}} */
 
+/* {{{ proto bool pspell_config_set(int conf, string option, string value)
+   set an option for this config */
+static PHP_FUNCTION(pspell_config_set)
+{
+        int type;
+        long conf;
+        char *option;
+        int option_len;
+        char *value;
+        int value_len;
+        PspellConfig *config;
+        
+        if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "lpp", &conf, &option, &option_len, &value, &value_len) == FAILURE) {
+                return;
+        }
+
+        PSPELL_FETCH_CONFIG;
+
+        if (php_check_open_basedir(value TSRMLS_CC)) {
+                RETURN_FALSE;
+        }
+
+        pspell_config_replace(config, option, value);
+
+        RETURN_TRUE;
+}
+/* }}} */
+
+
 /* {{{ proto bool pspell_config_ignore(int conf, int ignore)
    Ignore words <= n chars */
 static PHP_FUNCTION(pspell_config_ignore)
@@ -840,6 +877,7 @@
 	RETURN_TRUE;
 }
 
+
 /* {{{ proto bool pspell_config_personal(int conf, string personal)
    Use a personal dictionary for this config */
 static PHP_FUNCTION(pspell_config_personal)
